src/interface/types.rs

```diff
@@ -1038,8 +981,8 @@ pub fn get_constsigsett<'a>(union_argument: Arg) -> Result<Option<&'a SigsetType
     }
 }
 
-pub fn get_iovecstruct(union_argument: Arg) -> Result<*const interface::IovecStruct, i32> {
-    let data = unsafe { union_argument.dispatch_constiovecstruct };
+pub fn get_iovecstruct(argument: u64) -> Result<*const interface::IovecStruct, i32> {
```

Comment:
this file looks good!

my only comment is that maybe the param "argument" should be a bit clearer, maybe like "uncast_argument" or generic_argument